### ISSUE TITLE
taints are not taken into account by module beta-private-cluster-update-variant

### ISSUE THREAD

**Issue by sbehl27** [terraform-google-modules/terraform-google-kubernetes-engine#1325]
### TL;DR

When trying to create a GKE with a taint definition in variables using module/beta-private-cluster-update-variant, the taint settings are not taken into account. 
I thing the reason is, that the node-pool is created with a random_id suffix, but the taint is to be created for the node_pool without the random_id suffix. Hence it has no effect.

### Expected behavior

default-node-pool has a taint when specified in variables passed to terraform

can be checked with `kubectl get nodes -o json | jq '.items[].spec.taints'`

### Observed behavior

taint on node-pool is not created

### Terraform Configuration

```hcl
node_pools_taints = {
    all               = []
    default-node-pool = [
      {
        key    = "default-node-pool"
        value  = true
        effect = "PREFER_NO_SCHEDULE"
      },
    ]
  }
```

### Terraform Version

```sh
1.2.4
```

### Additional information

I think the issue is caused by line 568 in cluster.tf:
```
dynamic "taint" {
      for_each = concat(
        local.node_pools_taints["all"],
        local.node_pools_taints[each.value["name"]],
      )
      content {
        effect = taint.value.effect
        key    = taint.value.key
        value  = taint.value.value
      }
}
```

Instead of `each.value["name"]` also `{ for k, v in random_id.name : k => v.hex }[each.key]` should be used. Maybe working with a local could be helpful here.

**Comment by sbehl27** [comment]
sorry was my mistake
